source/linux/How_to_Guides/Target/How_to_fix_y2k38.rst

```diff
@@ -0,0 +1,43 @@
+.. include:: /_replacevars.rst
+
+How to Fix Y2K38 Issue
+======================
+
+What is the Y2K38 Issue?
+------------------------
+
+The Y2K38 issue is a bug which disallows a date from exceeding 03:14:07 UTC on January 19, 2038, on 32-bit systems. This
+is because the structures used to store the date experience an integer overflow when the date is set beyond the mentioned
+one. This causes the system to crash.
+For more information, see: https://en.wikipedia.org/wiki/Year_2038_problem
+
+There are new data structures that are used on 32-bit systems to store date/time in longer fields, thereby avoiding the
+overflow and the crash. However, since setting and displaying date/time uses multiple software components (such as the
+kernel, glibc etc), **all** code on a filesystem must be Y2K38-compliant (i.e. all code must be using the new data
+structures and functions) for the filesystem to be Y2K38-friendly. If any piece of software uses the older structures
+and functions, it makes the whole filesystem vulnerable.
+
+To understand how the problem was fixed in Scarthgap filesystem, refer:
+https://static.sched.com/hosted_files/osseu2024/8b/EOSS%20Vienna%202024%20-%20Surviving%20Y2038.pdf
+
+Building a Y2K38-friendly Image
+-------------------------------
+
+In SDK 9.3, TI's 32-bit platforms are still based on Kirkstone filesystem, which is NOT Y2K38-friendly. The Y2K38 issue
+is fixed in OE-core Scarthgap, on which the next release will be based.
+
+Until then, if a Y2K38-friendly filesystem is required, obtain a :file:`tisdk-thinlinux-image` from the latest
+`cicd.scarthgap.x`:
+
+.. ifconfig:: CONFIG_part_variant in ('AM335X')
+
+   https://software-dl.ti.com/cicd-report/linux/index.html?section=platform&platform=am335x
+
+.. ifconfig:: CONFIG_part_variant in ('AM437X')
+
+   https://software-dl.ti.com/cicd-report/linux/index.html?section=platform&platform=am437x
+
+The above images are based on the 6.6 Kernel and Scarthgap filesystem, and thus has the Y2K38 issue fixed.
+
+To build a Y2K38-friendly image on the 6.6 Kernel and Scarthgap filesystem baseline, follow the "Build Instructions"
+section present in the above link.
```

Comment:
One takeaway that I would like us to focus on a bit more here: TI can provide a base filesystem image that is y2038 friendly, but a customer can still introduce their own code that breaks y2038 compatibility. Customers need to be intentional about their software development, and about testing their code to ensure that the final product is still y2038.

Most of these concepts are already covered in the static.sched.com link above, so it is sufficient to add another paragraph discussing the above concept, and pointing to that same link for more information.

This information in particular is important for BOTH AM335x & AM437x, and AM57x customers (regardless of whether or not the out-of-the-box TI filesystem on AM57x is y2038-friendly or not)

Reply:
Addressed this in a "note" block, @nsaulnier-ti. Is that alright, or do you want it to be a paragraph?